lib/Dialect/FIRRTL/Transforms/LowerXMR.cpp

```diff
@@ -389,24 +405,35 @@ class LowerXMRPass : public LowerXMRBase<LowerXMRPass> {
       // attribute which specifies the internal path into the extern module.
       // This string attribute will be used to generate the final xmr.
       auto internalPaths = extRefMod.getInternalPaths();
-      // No internalPaths implies no RefType ports.
-      if (internalPaths.empty())
-        return success();
+      // Set internalPaths to macro?
       size_t pathsIndex = 0;
       auto numPorts = inst.getNumResults();
+      StringAttr abiInclude;
+      auto getPath = [&](size_t portNo) {
+        if (!internalPaths.empty())
+          return internalPaths[pathsIndex++].cast<StringAttr>();
+
+        auto [macro, filename] = getRefABIAttrs(extRefMod, portNo, true);
+        assert(!abiInclude || abiInclude == filename);
+        abiInclude = filename;
```

Comment:
This simple include approach has downsides and limitations: In the passthrough case (ref from extmodule forwarded out of the top module) the ref ABI file is no longer standalone-- e.g., will contain:
```
`define ref_DUT_DUT_read i.`ref_Inner_Inner_read
`define ref_DUT_DUT_write i.`ref_Inner_Inner_write
```

So either need to track which refs need which include's and change ABI such that the `ref_*.sv` files are responsible for containing the next-level `include`s (by name that the macros they use requires the information be stored under, here `ref_Inner_Inner.sv`, or we can handle this an entirely different way (such as w/o emitting includes ourselves).

Reply:
Discussed elsewhere, punt on includes.